src/core/Form/SearchInput/SearchInput.tsx

```diff
@@ -1,47 +1,265 @@
+        this.setState({ value: newValue });
+      }
+      if (propOnChange) {
+        propOnChange(newValue);
+      }
+    };
+
+    const onSearch = () => {
+      if (!!propOnSearch) {
+        propOnSearch(this.state.value);
+      }
+    };
+
+    const onClear = () => {
+      conditionalSetState('');
+      setTimeout(() => {
+        if (this.inputRef.current) {
+          this.inputRef.current.focus();
+        }
+      }, 100);
+    };
+
+    const onKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (!!this.state.value && event?.key === 'Enter') {
+        onSearch();
+      }
+    };
+
+    const searchButtonDerivedProps = {
+      ...searchButtonProps,
+      className: classnames(
+        searchButtonProps?.className,
+        searchInputClassNames.button,
+        searchInputClassNames.searchButton,
+      ),
+      ...(!!this.state.value
+        ? { onClick: onSearch }
+        : { tabIndex: -1, 'aria-hidden': true }),
+    };
+    const clearButtonProps = {
+      className: classnames(
+        searchInputClassNames.button,
+        searchInputClassNames.clearButton,
+      ),
+      ...(!!this.state.value
+        ? { onClick: onClear }
+        : { tabIndex: -1, 'aria-hidden': true }),
+    };
+
+    const getDescribedBy = () => {
+      if (statusText || ariaDescribedBy) {
+        return {
+          'aria-describedby': [
+            ...(statusText ? [this.statusTextId] : []),
+            ariaDescribedBy,
+          ].join(' '),
+        };
+      }
+      return {};
+    };
+
+    return (
+      <HtmlDiv
+        {...inputContainerProps}
+        className={classnames(className, baseClassName, {
+          [searchInputClassNames.error]: status === 'error',
+          [searchInputClassNames.notEmpty]: !!this.state.value,
+          [searchInputClassNames.fullWidth]: fullWidth,
+        })}
+      >
+        <LabelText htmlFor={this.id} labelMode={labelMode} as="label">
+          {labelText}
+        </LabelText>
+        <HtmlDiv className={searchInputClassNames.functionalityContainer}>
+          <HtmlDiv className={searchInputClassNames.inputElementContainer}>
+            <HtmlInputWithRef
+              {...passProps}
+              {...getDescribedBy()}
+              forwardRef={this.inputRef}
+              aria-invalid={status === 'error'}
+              id={this.id}
+              className={searchInputClassNames.inputElement}
+              type="search"
+              role="searchbox"
+              value={this.state.value}
+              placeholder={visualPlaceholder}
+              onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                conditionalSetState(event.currentTarget.value);
+              }}
+              onKeyPress={onKeyPress}
+            />
+          </HtmlDiv>
+          <HtmlButton {...clearButtonProps}>
+            <VisuallyHidden>{clearButtonLabel}</VisuallyHidden>
+            <Icon
+              aria-hidden={true}
+              icon="close"
```

Comment:
Not issue of this PR, but should we have separate icons for clear and close?

Reply:
Maybe, but I guess it's the same one in the designs as well.

Reply:
Yes, let's leave those like currently and adjust later if new icons will come for both.